airflow/providers/amazon/aws/fs/s3.py

```diff
@@ -49,7 +50,7 @@ def get_fs(conn_id: str | None, storage_options: dict[str, str] | None = None) -
     try:
         from s3fs import S3FileSystem
     except ImportError:
-        raise ImportError(
+        raise AirflowOptionalProviderFeatureException(
```

Comment:
This change not related and could be done by the separate PR. I will revert it back